### PR TITLE
chore: PHPStan 11 array rules

### PR DESCRIPTION
**Pull request by keradus**
shall we enable?
(eg enable with exception counter).
it's like ~900 violations in current codebase.

### CODE REVIEW

**Comment by keradus** [comment]
@localheinz @julienfalque @kubawerlos @SpacePossum @Wirone 
would like to hear your voice, if you have opinion on that

**Review by localheinz — APPROVED** [review]

**Comment by mvorisek** [comment]
I do not think this is good idea until https://github.com/phpstan/phpstan/issues/8438 is implemented.

**Comment by Wirone** [comment]
@keradus what would be the real added value when we enable this? I believe we currently enforce proper array shapes for new code, and have already errors dump in the baseline. Enabling this option would probably make it even stricter, but I would rather prefer resolving root problems like fixers' config shapes, then would go into stricter analysis. Can you provide examples of what currently is not reported by tests/SA, which would be reported when these options are enabled?

**Comment by keradus** [comment]
config shapes hopefully resolved in #7988

**Comment by keradus** [comment]
I enabled `reportPossiblyNonexistentConstantArrayOffset`, only few violations after #7988

I keep `reportPossiblyNonexistentGeneralArrayOffset` as future opportunity - it's worth it IMHO: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8089

File: phpstan.dist.neon [hunk 1]
```diff
@@ -13,6 +13,8 @@ parameters:
         - tests/Fixtures
     polluteScopeWithLoopInitialAssignments: true # Do not enforce assignments outside of the loops
     reportUnmatchedIgnoredErrors: true # Do not allow outdated errors in the baseline
+    reportPossiblyNonexistentGeneralArrayOffset: true # enabled, but huge exceptions count; example fix: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8089
+    reportPossiblyNonexistentConstantArrayOffset: true
```
Comment (keradus) [inline phpstan.dist.neon:17]: when we opened the PR we had ~900 violations. now it's only 560 in total

File: phpstan.dist.neon [hunk 2]
```diff
@@ -52,5 +54,9 @@ parameters:
             message: '#^Method PhpCsFixer\\Tests\\.+::provide.+Cases\(\) return type has no value type specified in iterable type iterable\.$#'
             path: tests
             count: 1013
+        -
+            identifier: offsetAccess.notFound
+            path: .
+            count: 560
```
Comment (keradus) [inline phpstan.dist.neon:60]:
```suggestion
            count: 559
```

**Pull request merged**